src/units/abilities.cpp

```diff
@@ -1169,7 +1169,7 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 }
 
 //Functions used for change damage_type list with damage
-static std::vector<std::string> damage_type_list(const unit_ability_list& abil_list, const std::string& type)
+static std::optional<std::string> damage_type_list(const unit_ability_list& abil_list, const std::string& type)
```

Comment:
Maybe rename it just `damage_type`, as it no longer returns a list, and it's just a helper function for the other `damage_type` function.

Reply:
rename to damage_type_helper